connect_ext_ppr/models/configuration.py

```diff
@@ -0,0 +1,32 @@
+from datetime import datetime
+
+import sqlalchemy as db
+
+from connect_ext_ppr.db import Model
+from connect_ext_ppr.models.deployment import Deployment
+from connect_ext_ppr.models.enums import ConfigurationStateChices
+from connect_ext_ppr.models.file import File
+
+
+class Configuration(Model):
+    __tablename__ = 'configurations'
+
+    PREFIX = 'CFL'
+
+    id = db.Column(db.String(20), primary_key=True)
+    file = db.Column(db.ForeignKey(File.id))
+    deployment = db.Column(db.ForeignKey(Deployment.id))
+    state = db.Column(
+        db.Enum(ConfigurationStateChices, validate_strings=True),
+        default=ConfigurationStateChices.INACTIVE,
+    )
+    created_at = db.Column(db.DateTime(), default=datetime.utcnow)
+    created_by = db.Column(db.String(20))
+    updated_at = db.Column(db.DateTime(), default=datetime.utcnow)
+    updated_by = db.Column(db.String(20))
+
+    def activate(self):
+        self.state = ConfigurationStateChices.ACTIVE
+
+    def deleted(self):
```

Comment:
[COD-C] Maybe better `delete` as it is an action, WDYT?

Reply:
If you refer to the article it said that for deletion, should be a mechanism of fake delete, that will mark configuration file as deleted, because it could be related to an "in use" PPR build on top of it, so my method is more like "mark as deleted", not delete like the ORM will do.